# 프라미스와 에러 핸들링

[https://ko.javascript.info/promise-error-handling](https://ko.javascript.info/promise-error-handling)

.catch는 여러 개의 .then뒤에 올 수 있다. 따라서 마지막에 .catch가 이용되는 경우가 많다. 

## 암시적 try...catch

throw new Error()는 reject(new Error())와 같음.

## 다시 던지기

.catch이후 .then핸들러를 이용하여 오류를 처리할 수 있다. 

## 처리되지 못한 거부

만약 .catch가 없다면 어떻게 에러가 처리될까? 스크립트가 죽고 콘솔창에 메세지가 출력될 것이다. 

## 요약

* .catch는 프라미스에서 발생한 모든 에러를 다룬다. reject()가 호출되거나 에러가 던저지면 .catch가 이를 해결한다. 
* .catch는 에러를 처리하고 싶은 지점에 정확히 위치해야 한다. 
* 에러 발생 시 회복 방법이 없다면 .catch를 사용하지 않아도 된다. 

## 과제.

이게 왜 이렇게 되는지 이해를 못하겠어요.